homeassistant/components/mqtt/mixins.py

```diff
@@ -868,7 +872,37 @@ async def async_removed_from_registry(self) -> None:
     @callback
     def add_to_platform_abort(self) -> None:
         """Abort adding an entity to a platform."""
+        hook_register: dict[tuple, Callable] = self.hass.data[
+            DATA_MQTT_DISCOVERY_REGISTRY_HOOKS
+        ]
+
+        async def _async_clear_discovery_topic_if_entity_removed(
```

Comment:
I think we should move this up to module level to allow the entity to be garbage collected.